core/packages/contracts/src/Assets.sol

```diff
@@ -48,16 +49,17 @@ library Assets {
 
         _transferToAgent(assetHubAgent, token, sender, amount);
         if (destinationChain == assetHubParaID) {
-            payload = SubstrateTypes.SendToken(assetHubAgent, token, destinationAddress, amount);
+            payload = SubstrateTypes.SendToken(gateway, token, destinationAddress, amount);
         } else {
-            payload = SubstrateTypes.SendToken(assetHubAgent, token, destinationChain, destinationAddress, amount);
+            payload = SubstrateTypes.SendToken(gateway, token, destinationChain, destinationAddress, amount);
         }
         extraFee = $.sendTokenFee;
 
         emit TokenSent(sender, token, destinationChain, abi.encodePacked(destinationAddress), amount);
     }
 
     function sendToken(
+        address gateway,
```

Comment:
See my comment below.

```suggestion
```

Reply:
addressed in https://github.com/Snowfork/snowbridge/pull/904/commits/8289362594510c9672b87bd44bfbf09dcbe3f596